oop-relationships/composition.md

```diff
@@ -15,11 +15,192 @@
 
 ## Composition Affects Our Attributes
```

Comment:
I appreciate that you want to give motivation to this, but considering that this is the first part of the lesson, I'd like to try reworking these paragraphs and focusing on delivering syntax?

In this specific case, I'd probably rather a "show syntax, then give explanation to students about why it makes sense" since this motivation (of why composition == initializing attributes in constructor) is farther from non-programming logic than a different subject.

I personally don't feel confident that the phrase "simple composition relationships" and the example "a composite class might initialize its attributes directly in its constructor. This is common ..." would come across as simple/obvious/common? Obviously it'd be nice if this weren't the case, but I'm not confident that the curriculum has enough "make up a class and pick the attributes in the constructor" exercises to get them to practice... or at least, readily realize that we can initialize an attirbute to an empty string, list, etc. I personally think the bridge you've set up relies on learning that has sunk in/been synthesized to this level yet.

Also, did your suggestion "it's more common to set the components to instances provided by other code that uses our class" mean to be "it's more common to set the attributes to instances provided by other code that uses our class"

```suggestion
## Initializing Components as Attributes in the Constructor

It's common to set the attributes of a composite class to component instances provided by other code that uses our class. Python lets us assign to attributes directly. We can let all the components in our class be passed into the constructor of the composite class, and assign the attributes in the constructor itself.

Consider this example:
```

Reply:
It was a way of saying that an instance doesn't usually know the context it's being used in, and relies on the code using it to provide that context through its initialization.  Apparently it was so opaque that I had to reread it a few times before I was sure what I meant myself.

I like the pared down version.  I'm tempted to recommend leaving the word 'component' out, except that it's sort of a keyword in the section.  It's just that 'attributes of a composite class to component instances' feels really jargon-dense to me.  But even as is, the suggestion is an improvement.